### ISSUE TITLE
some tests file leave stuff under /tmp

### ISSUE THREAD

**Issue by iabyn** [dagolden/CPAN-Meta-YAML#2]
On blead, some of the CPAN-Meta-YAML test files leave temporary files under /tmp.
The following patch seems to fix it:

```
diff --git a/cpan/CPAN-Meta-YAML/t/12_write.t b/cpan/CPAN-Meta-YAML/t/12_write.t
index 8c7b4be..25f6298 100644
--- a/cpan/CPAN-Meta-YAML/t/12_write.t
+++ b/cpan/CPAN-Meta-YAML/t/12_write.t
@@ -44,7 +44,8 @@ for my $c ( @cases ) {
         @warnings = ();
 
         # get a tempfile name to write to
-        my ($fh, $tempfile) = tempfile("YAML-Tiny-test-XXXXXXXX", TMPDIR => 1 );
+        my ($fh, $tempfile) = tempfile("YAML-Tiny-test-XXXXXXXX",
+                                    TMPDIR => 1, UNLINK => 1);
         my $short_tempfile = basename($tempfile);
         close $fh; # avoid locks on windows
 
diff --git a/cpan/CPAN-Meta-YAML/t/21_yamlpm_compat.t b/cpan/CPAN-Meta-YAML/t/21_yamlpm_compat.t
index e6a69d3..2038de5 100644
--- a/cpan/CPAN-Meta-YAML/t/21_yamlpm_compat.t
+++ b/cpan/CPAN-Meta-YAML/t/21_yamlpm_compat.t
@@ -31,7 +31,7 @@ use CPAN::Meta::YAML;
     my $arrayref = [ 1 .. 5 ];
     my $hashref = { alpha => 'beta', gamma => 'delta' };
 
-    my ($fh, $filename) = tempfile;
+    my ($fh, $filename) = tempfile(UNLINK => 1);
     close $fh; # or LOCK_SH will hang
 
     my $rv = CPAN::Meta::YAML::DumpFile(
```

**Comment by jkeenan** [comment]
David, if I read https://github.com/Perl-Toolchain-Gang/YAML-Tiny/commit/82862dfc94f71ceb75d5e9effb0dd751e9367eba correctly, it appears that the solution implemented upstream in YAML-Tiny was to use functionality from Test::TempDir::Tiny (with which I believe you are acquainted :-) ).

However, my understanding is that CPAM-Meta-YAML is intended for inclusion in the Perl 5 core.  If we were to use the upstream solution, we would have to bring Test::TempDir::Tiny into the core distribution.  I can't recall any discussion of that, and it's rather late in this year's release cycle to contemplate doing that.

I stumbled on this problem tonight while repeatedly running 'make test_harness' in core.  I experimented and say that simply changing 'TMPDIR' to 'UNLINK' in line 47 of cpan/CPAN-Meta-YAML/t/12_write.t prevented retention of the temporary files in /tmp.

Thank you very much.
Jim Keenan

**Comment by dagolden** [comment]
Yeah, I think it really needs to vendor Test::TempDir::Tiny into the t/lib
directory.

On Sat, Jan 31, 2015 at 10:42 PM, James E Keenan <notifications@github.com>
wrote:

> David, if I read Perl-Toolchain-Gang/YAML-Tiny@82862df
> <https://github.com/Perl-Toolchain-Gang/YAML-Tiny/commit/82862dfc94f71ceb75d5e9effb0dd751e9367eba>
> correctly, it appears that the solution implemented upstream in YAML-Tiny
> was to use functionality from Test::TempDir::Tiny (with which I believe you
> are acquainted :-) ).
>
> However, my understanding is that CPAM-Meta-YAML is intended for inclusion
> in the Perl 5 core. If we were to use the upstream solution, we would have
> to bring Test::TempDir::Tiny into the core distribution. I can't recall any
> discussion of that, and it's rather late in this year's release cycle to
> contemplate doing that.
>
> I stumbled on this problem tonight while repeatedly running 'make
> test_harness' in core. I experimented and say that simply changing 'TMPDIR'
> to 'UNLINK' in line 47 of cpan/CPAN-Meta-YAML/t/12_write.t prevented
> retention of the temporary files in /tmp.
>
> Thank you very much.
> Jim Keenan
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dagolden/CPAN-Meta-YAML/issues/2#issuecomment-72349932>
> .
>



-- 
David Golden <dagolden@cpan.org> Twitter/IRC: @xdg

**Comment by karenetheridge** [comment]
https://github.com/Perl-Toolchain-Gang/YAML-Tiny/issues/25 - I'm getting this released one way or another tonight.

**Comment by karenetheridge** [comment]
YAML-Tiy-1.65 contains this fix.

**Comment by jkeenan** [comment]
Am looking forward to a new release of CPAN::Meta::YAML based on this.  Thanks.

**Comment by dagolden** [comment]
It's on my list of things todo tonight or tomorrow.

On Sat, Mar 14, 2015 at 10:41 PM, James E Keenan <notifications@github.com>
wrote:

> Am looking forward to a new release of CPAN::Meta::YAML based on this.
> Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dagolden/CPAN-Meta-YAML/issues/2#issuecomment-80812513>
> .
>



-- 
David Golden <dagolden@cpan.org> Twitter/IRC: @xdg

**Comment by dagolden** [comment]
CPAN-Meta-YAML-0.013 released to CPAN based on YAML-Tiny-1.65
